modular_zubbers/code/modules/job_interns/job_interns.dm

```diff
@@ -0,0 +1,269 @@
+		if(EXP_TYPE_SERVICE)
+			config_type = /datum/config_entry/number/intern_threshold_service
+		if(EXP_TYPE_MEDICAL)
+			config_type = /datum/config_entry/number/intern_threshold_medical
+		if(EXP_TYPE_SCIENCE)
+			config_type = /datum/config_entry/number/intern_threshold_science
+	if(!config_type)
+		return 0
+	return global.config.Get(config_type) * 60
+
+/// Returns whether a player should be tagged as an intern in this job
+/datum/job/proc/player_joins_as_intern(client/player_client)
+	if(!can_be_intern)
+		return FALSE
+	if(!CONFIG_GET(flag/allow_intern_job_tags))
+		return FALSE
+	if(!SSdbcore.Connect())
+		return FALSE
+	var/player_allows_intern = player_client.prefs.read_preference(/datum/preference/toggle/be_intern)
+	if(!player_allows_intern)
+		return FALSE
+	var/required_time
+	var/playtime
+	if(internship_use_self_exp_type)
+		var/list/play_records = player_client.prefs.exp
+		playtime = play_records[title] ? text2num(play_records[title]) : 0
+		required_time = get_intern_time_threshold()
+	else if(CONFIG_GET(flag/use_intern_master_job_unlock_threshold) && intern_master_job)
+		var/datum/job/master_job = SSjob.type_occupations[intern_master_job]
+		playtime = player_client.calc_exp_type(master_job.get_exp_req_type())
+		required_time = master_job.get_exp_req_amount()
+	else
+		if(!internship_dep_exp_type)
+			return FALSE
+		required_time = get_intern_time_threshold()
+		playtime = player_client.calc_exp_type(internship_dep_exp_type)
+	if(playtime >= required_time)
+		return FALSE
+	return TRUE
+
+/obj/item/card/id
+	var/intern_status = FALSE
+
+/// Sets an intern status and updates the label of the ID
+/obj/item/card/id/proc/set_intern_status(new_status)
+	intern_status = new_status
+	update_label()
+
+/obj/item/card/id/proc/get_job_title()
+	var/assignment_string
+	if(is_intern || intern_status)
+		if(assignment)
+			assignment_string = trim?.intern_alt_name || "Intern [assignment]"
+		else
+			assignment_string = "Intern"
+	else
+		assignment_string = assignment
+	return assignment_string
+
+// Command
+/datum/job/captain
+	internship_dep_exp_type = EXP_TYPE_COMMAND
+
+/datum/job/chief_engineer
+	internship_dep_exp_type = EXP_TYPE_ENGINEERING
+
+/datum/job/chief_medical_officer
+	internship_dep_exp_type = EXP_TYPE_MEDICAL
+
+/datum/job/head_of_security
+	internship_dep_exp_type = EXP_TYPE_SECURITY
+
+/datum/job/head_of_personnel
+	internship_dep_exp_type = EXP_TYPE_SERVICE
+
+/datum/job/research_director
+	internship_dep_exp_type = EXP_TYPE_SCIENCE
+
+/datum/job/quartermaster
+	internship_dep_exp_type = EXP_TYPE_SUPPLY
+
+/datum/job/blueshield
+	internship_dep_exp_type = EXP_TYPE_COMMAND
+
+/datum/job/nanotrasen_consultant
+	internship_dep_exp_type = EXP_TYPE_COMMAND
+
+// Security
+/datum/job/warden
+	internship_dep_exp_type = EXP_TYPE_SECURITY
+	intern_master_job = /datum/job/head_of_security
+
+/datum/job/security_officer
+	internship_dep_exp_type = EXP_TYPE_SECURITY
+	intern_master_job = /datum/job/head_of_security
```

Comment:
Ditto issue for `intern_master_job`, this information is already present in `department_head`